newIDE/app/src/AssetStore/PrivateAssetPackDialog.js

```diff
@@ -0,0 +1,244 @@
+// @flow
+import * as React from 'react';
+import { I18n } from '@lingui/react';
+import { type PrivateAssetPackListingData } from '../Utils/GDevelopServices/Shop';
+import {
+  getPrivateAssetPackDetails,
+  type PrivateAssetPackDetails,
+} from '../Utils/GDevelopServices/Asset';
+import Text from '../UI/Text';
+import { t, Trans } from '@lingui/macro';
+import Dialog from '../UI/Dialog';
+import TextButton from '../UI/TextButton';
+import AlertMessage from '../UI/AlertMessage';
+import PlaceholderLoader from '../UI/PlaceholderLoader';
+import { ResponsiveLineStackLayout } from '../UI/Layout';
+import { Column, LargeSpacer, Line, Spacer } from '../UI/Grid';
+import {
+  getUserPublicProfile,
+  type UserPublicProfile,
+} from '../Utils/GDevelopServices/User';
+import PublicProfileDialog from '../Profile/PublicProfileDialog';
+import Link from '../UI/Link';
+import Mark from '../UI/CustomSvgIcons/Mark';
+import Cross from '../UI/CustomSvgIcons/Cross';
+import { Paper } from '@material-ui/core';
+import ImagesDisplay from '../UI/ImagesDisplay';
+import { useResponsiveWindowWidth } from '../UI/Reponsive/ResponsiveWindowMeasurer';
+
+type Props = {|
+  privateAssetPack: PrivateAssetPackListingData,
+  onClose: () => void,
+|};
+
+const sortedContentType = [
+  'sprite',
+  '9patch',
+  'tiled',
+  'particleEmitter',
+  'font',
+  'audio',
+  'partial',
+];
+
+const contentTypeToMessageDescriptor = {
+  sprite: t`Sprites`,
+  '9patch': t`Panel sprites`,
+  tiled: t`Tiled sprites`,
+  particleEmitter: t`Particle emitters`,
+  font: t`Fonts`,
+  audio: t`Audios`,
+  partial: t`Other`,
+};
+
+const PrivateAssetPackDialog = ({
+  privateAssetPack: { id, name, description, sellerId },
+  onClose,
+}: Props) => {
+  const [
+    assetPackDetails,
+    setAssetPackDetails,
+  ] = React.useState<?PrivateAssetPackDetails>(null);
+  const [isFetchingDetails, setIsFetchingDetails] = React.useState<boolean>(
+    false
+  );
+  const [
+    openSellerPublicProfileDialog,
+    setOpenSellerPublicProfileDialog,
+  ] = React.useState<boolean>(false);
+  const [
+    sellerPublicProfile,
+    setSellerPublicProfile,
+  ] = React.useState<?UserPublicProfile>(null);
+  const [errorText, setErrorText] = React.useState<?React.Node>(null);
+  const windowWidth = useResponsiveWindowWidth();
+
+  React.useEffect(
+    () => {
+      (async () => {
+        setIsFetchingDetails(true);
+        try {
+          const details = await getPrivateAssetPackDetails(id);
+          const profile = await getUserPublicProfile(sellerId);
+          setAssetPackDetails(details);
+          setSellerPublicProfile(profile);
+        } catch (error) {
+          if (error.response && error.response.status === 404) {
+            setErrorText(
+              <Trans>
+                Asset pack not found - An error occurred, please try again
+                later.
+              </Trans>
+            );
+          } else {
+            setErrorText(<Trans>Unknown error</Trans>);
```

Comment:
maybe improve that message for the user? something like `An error occurred, please try again later`